operators/advanced_network/advanced_network/managers/rivermax/rivermax_service/CMakeLists.txt

```diff
@@ -0,0 +1,100 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+cmake_minimum_required(VERSION 3.20)
+
+project(rivermax_service)
+
+add_library(${PROJECT_NAME} SHARED
+  #rmax_base_service.cpp
+  #rmax_ipo_receiver_service.cpp
+  #ipo_receiver_io_node.cpp
+  # We'll add the rmax_ipo_receiver.cpp after fetching it
+)
+
+target_include_directories(${PROJECT_NAME} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+
+target_compile_features(${PROJECT_NAME}
+  PUBLIC cxx_std_17
+)
+target_compile_definitions(${PROJECT_NAME} PUBLIC RMAX_APPS_LIB_FLAT_STRUCTURE)
+
+set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+
+# Set RMAX_CUDA flag
+set(RMAX_CUDA TRUE)
+
+# Check if the environment variable RMAX_TEGRA is set
+if(NOT DEFINED ENV{RMAX_TEGRA})
+    message(STATUS "Environment variable RMAX_TEGRA was not defined. Defaulting to FALSE")
+    set(RMAX_TEGRA FALSE)
+else()
+    # Convert the environment variable to CMake boolean TRUE or FALSE
+    if("$ENV{RMAX_TEGRA}" STREQUAL "TRUE" OR "$ENV{RMAX_TEGRA}" STREQUAL "ON")
+        set(RMAX_TEGRA TRUE)
+    else()
+        set(RMAX_TEGRA FALSE)
+    endif()
+endif()
+
+unset(RMAX_TEGRA CACHE)
+message(STATUS "RMAX_TEGRA is set to: ${RMAX_TEGRA}")
+
+if(RMAX_TEGRA)
+    target_compile_definitions(${PROJECT_NAME} PUBLIC RMAX_TEGRA=1)
+else()
+    target_compile_definitions(${PROJECT_NAME} PUBLIC RMAX_TEGRA=0)
+endif()
```

Comment:
What is the point of all these? Do you need the libraries consuming the rdk to have these defined (RMAX_TEGRA, RMAX_CUDA, RMAX_APPS_LIB_FLAT_STRUCTURE)